CHANGELOG.md

```diff
@@ -25,9 +25,15 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 
 ## API changes
 
-### The `type_list` header has moved
+### Argument parser
 
-If you included `<seqan3/core/type_list.hpp>` you need to change the path to `<seqan3/core/type_list/type_list.hpp>`.
+* The signature of input file and output file validator have been modified to template classes.
+  It is now possible to get the list of file extensions from the type of a seqan input file, respectively output file. Old code will likely not be affected unless the type was used explicitly.
```

Comment:
This is mixing up the description of a feature and an API Change. Also `Old code will likely not be affected` does not help people in actually fixing it, if it is affected. I think the API Changes section should provide at most minimal context and specify exactly what to do to make code compile.

I would propose to write:
```md

## New features

### Argument Parser: file validators accept files as template arguments

Passing a file as template argument to a file validator, e.g. `seqan3::input_file_validator<seqan3::sequence_file_input<>> validator{}` will automatically set all supported extensions.

[...]

## API Changes

### Argument Parser: file validators are now class templates

Most code should be unaffected; places that are affected need to explicitly provide empty template arguments, e.g. `seqan3::input_file_validator<>` instead of `seqan3::input_file_validator`.
```